### PR TITLE
chore: upgrade codebase to 3.6+

### PR DESCRIPTION
**Pull request by branchvincent**
Relates-to: python-poetry/poetry#3860

- [ ] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

In the same spirit as python-poetry/poetry#3988, this:
* Applies `pyupgrade --py36-plus` to the codebase
* Adds the above to `.pre-commit-config.yaml`

### CODE REVIEW

**Comment by kamyar** [comment]
@branchvincent any specific reason to reopen same change when there is already #3988 ?

**Comment by branchvincent** [comment]
> @branchvincent any specific reason to reopen same change when there is already #3988 ?

hey @kamyar! your pr is for [poetry](https://github.com/python-poetry/poetry) while this is for [poetry-core](https://github.com/python-poetry/poetry-core)

**Comment by kamyar** [comment]
Sorry missed that 🤦 😞 
Thank you for pointing it out

File: poetry/core/packages/dependency.py [hunk 1]
```diff
@@ -187,14 +187,14 @@ def base_pep_508_name(self) -> str:
 
         if isinstance(self.constraint, VersionUnion):
             if self.constraint.excludes_single_version():
-                requirement += " ({})".format(str(self.constraint))
+                requirement += f" ({str(self.constraint)})"
             else:
                 constraints = self.pretty_constraint.split(",")
                 constraints = [parse_constraint(c) for c in constraints]
                 constraints = [str(c) for c in constraints]
                 requirement += " ({})".format(",".join(constraints))
```
Comment (Leonardofreua) [inline poetry/core/packages/dependency.py:195]:
```suggestion
                requirement += f" ({','.join(constraints)})"
```

File: poetry/core/packages/dependency.py [hunk 2]
```diff
@@ -187,14 +187,14 @@ def base_pep_508_name(self) -> str:
...
         elif isinstance(self.constraint, Version):
-            requirement += " (=={})".format(self.constraint.text)
+            requirement += f" (=={self.constraint.text})"
         elif not self.constraint.is_any():
             requirement += " ({})".format(str(self.constraint).replace(" ", ""))
```
Comment (Leonardofreua) [inline poetry/core/packages/dependency.py:199]:
```suggestion
            requirement += f' ({str(self.constraint).replace(" ", "")})'
```

File: poetry/core/packages/dependency.py [hunk 3]
```diff
@@ -269,10 +269,10 @@ def to_pep_508(self, with_extras: bool = True) -> str:
                 requirement += " "
 
             if len(markers) > 1:
-                markers = ["({})".format(m) for m in markers]
+                markers = [f"({m})" for m in markers]
                 requirement += "; {}".format(" and ".join(markers))
```
Comment (Leonardofreua) [inline poetry/core/packages/dependency.py:273]:
```suggestion
                requirement += f"; {' and '.join(markers)}"
```

File: poetry/core/packages/directory_dependency.py [hunk 4]
```diff
@@ -123,7 +123,7 @@ def base_pep_508_name(self) -> str:
             requirement += "[{}]".format(",".join(self.extras))
```
Comment (Leonardofreua) [inline poetry/core/packages/directory_dependency.py:123]:
```suggestion
            requirement += f"[{','.join(self.extras)}]"
```

File: poetry/core/packages/file_dependency.py [hunk 5]
```diff
@@ -108,7 +108,7 @@ def base_pep_508_name(self) -> str:
             requirement += "[{}]".format(",".join(self.extras))
```
Comment (Leonardofreua) [inline poetry/core/packages/file_dependency.py:108]:
```suggestion
            requirement += f"[{','.join(self.extras)}]"
```

File: poetry/core/packages/package.py [hunk 6]
```diff
@@ -152,7 +152,7 @@ def full_pretty_version(self) -> str:
 
         # if source reference is a sha1 hash -- truncate
         if len(self.source_reference) == 40:
-            return "{} {}".format(self._pretty_version, self.source_reference[0:7])
+            return f"{self._pretty_version} {self.source_reference[0:7]}"
 
         return "{} {}".format(
```
Comment (Leonardofreua) [inline poetry/core/packages/package.py:157]: It missing to apply the f-string in the last return of the method `full_pretty_version()`.

File: poetry/core/packages/url_dependency.py [hunk 7]
```diff
@@ -48,7 +48,7 @@ def base_pep_508_name(self) -> str:
         if self.extras:
             requirement += "[{}]".format(",".join(self.extras))
```
Comment (Leonardofreua) [inline poetry/core/packages/url_dependency.py:49]:
```suggestion
            requirement += f"[{','.join(self.extras)}]"
```

File: poetry/core/packages/vcs_dependency.py [hunk 8]
```diff
@@ -108,7 +108,7 @@ def base_pep_508_name(self) -> str:
             requirement += "[{}]".format(",".join(self.extras))
```
Comment (Leonardofreua) [inline poetry/core/packages/vcs_dependency.py:108]:
```suggestion
            requirement += f"[{','.join(self.extras)}]"
```

File: poetry/core/packages/vcs_dependency.py [hunk 9]
```diff
@@ -108,7 +108,7 @@ def base_pep_508_name(self) -> str:
...
 
         if parsed_url.protocol is not None:
-            requirement += " @ {}+{}@{}".format(self._vcs, self._source, self.reference)
+            requirement += f" @ {self._vcs}+{self._source}@{self.reference}"
         else:
             requirement += " @ {}+ssh://{}@{}".format(
```
Comment (Leonardofreua) [inline poetry/core/packages/vcs_dependency.py:113]: It missing to apply the f-string on line 113.

File: poetry/core/version/pep440/version.py [hunk 10]
```diff
@@ -120,7 +120,7 @@ def to_string(self, short=False):
 
         if self.epoch:
             # if epoch is non-zero we should include it
-            version_string = "{}!{}".format(self.epoch, version_string)
+            version_string = f"{self.epoch}!{version_string}"
 
         if self.local:
             version_string += "+{}".format(".".join(map(str, self.local)))
```
Comment (Leonardofreua) [inline poetry/core/version/pep440/version.py:126]:
```suggestion
            version_string += f"+{'.'.join(map(str, self.local))}"
```

File: poetry/core/version/requirements.py [hunk 11]
```diff
@@ -92,18 +92,18 @@ def __str__(self) -> str:
         parts = [self.name]
 
         if self.extras:
-            parts.append("[{0}]".format(",".join(sorted(self.extras))))
+            parts.append("[{}]".format(",".join(sorted(self.extras))))
```
Comment (Leonardofreua) [inline poetry/core/version/requirements.py:95]:
```suggestion
            parts.append(f"[{','.join(sorted(self.extras))}]")
```

**Review by Leonardofreua — CHANGES_REQUESTED** [review]

**Comment by branchvincent** [comment]
closing in favor of #230

**Pull request closed without merging**